### ISSUE TITLE
Some bugs ?

### ISSUE THREAD

**Issue by lorenzolamasse** [malacalypse/NordLead3Librarian#1]
Hello there,

This project seems a bit on hold, and I know you don't spend time on it, but here is a feedback, it seems that it's currently unuseable alas:
When I initiate a request for a bank, the software seems to ask for performances, and goes seemingly randomly through different stuff (performance, various banks, ...), I can't figure out what it is actually doing, and anyway my synth doesn't send anything unless I initiate a manual dump, in which case the soft is somehow able to grab them even if they are not those requested...
I also get error messages at random: The operation couldn’t be completed. (Nord_Lead_3_Librarian.DataError error 1.)
Audition doesn't seem to work at all (probably linked to the issue where the NL doesn't seems to get the requests properly).
It's impossible to target a specific SysEX ID, no big deal but if people happen to have several stuff from Clavia that could be a problem.

Anyway, if you happen not to have enough time to work on it (you have stated elsewhere that you wouldn't do any support) you might consider opening the source code so people can join in and help at least correcting the issues, as it's too bad this goes to abandonware ?

Cheers,

**Comment by malacalypse** [comment]
Hi @lorenzolamasse , 

It sounds like you have a MIDI issue - are you trying to use this over an RTP MIDI connection such as with an iConnectivity device? Unfortunately those have too small of a sysex buffer and do not transmit the whole message, causing issues with sysex across a variety of synths with larger-format data dumps. 

I use this app frequently enough myself to recognize your symptoms as being related to this buffer issue, and they do not occur with normal "dumb" USB-to-MIDI adapters. I did not have any issues with older iConnectivity MIDI interfaces, as long as they were connected to the computer by USB, either, but only with RTP were they problematic. 

As for the comment on targeting a specific sysex ID - this software will only talk to an NL3 so unless you have multiple NL3s on the same MIDI port (why?) this should work perfectly fine if you have other non-NL3 devices on the same port, since the SysEx unit byte is different. Use a different MIDI port for each NL3 (you should be doing this anyway for bandwidth - 4x multitimbral synths should not be sharing ports with anything else) and you won't have an issue. 

I won't be open sourcing the software, sorry. 

If you can provide more details on your exact setup I may be able to shed some more light on your issue.

**Comment by malacalypse** [comment]
Please provide:
- Model and connection type of your USB interface
- An overview of your MIDI setup (which devices, in which order, on which ports, connected how - this matters)
- What version of firmware is on your NL3
- The values of PrgChg, MIDICtrl, and InstrSysexID for your NL3
- Which Mac and OS version you're using

**Comment by malacalypse** [comment]
This issue is pending closure in 5 days due to inactivity.

**Comment by lorenzolamasse** [comment]
Yes, I need to change MIDI connections to try your suggestion, as I was indeed using a low end interface, but in the mean time I've totally moved the studio ;-)

**Comment by lorenzolamasse** [comment]
With the proper Audio Interface I managed to gather the performances and patches, thanks for the tip !
Sending banks works fine as well.
Got an issue still, patch renaming that was initially fine abruptly stopped working, and I cannot use it anymore, even after a computer reboot, the changes I make are not saved anymore in the display when pressing enter. Is there any file somewhere, should I try to reinstall ?
As mentioned above, when I try to fetch one bank it still tries to fetch all performances and some other banks as well, only a full fetch seems to work right.
Thanks for any clue !

**Comment by malacalypse** [comment]
Ok, glad it works. 

I'll see if I can find any time to take a look at it. The single bank fetch should work, so if it doesn't that's a bug. 

I'll see if I have any issues with the naming as well. Does this happen the first time you try to name a patch? Do you have international characters in the names you're trying to use? Are you using a non-US keyboard and character set? The app cannot work with international characters as the Nord can't either. 

Can you also please give me the model of Mac (exactly) and OS version you're using?

**Comment by lorenzolamasse** [comment]
Yeah sure, I'm running Mojave 10.14.16 on a iMac from 2017.
I'm not using any international characters, but running on a French keyboard and layout. The weird thing is the fact it did work at first, and suddenly stopped working :/

**Comment by malacalypse** [comment]
Hm, that's certainly odd. Did it stop working in the middle of using it? Or was it working one time the app was open and when you closed it, it stopped working immediately on opening it again? I'll try to replicate, but the more details I get the easier it'll be. And I can't promise when I'll be able to get to it, sadly.

**Comment by lorenzolamasse** [comment]
Right during an edit session. Maybe I sent a bank to the synth and right after that it stopped working... If it comes back at some point I'll try to reproduce...

**Comment by lorenzolamasse** [comment]
I noticed as well, the "Send to active slot on enter" is grayed out, I don't think it was initially...
Is there any settings file I could remove to start over ?

**Comment by malacalypse** [comment]
No, there are no settings that it stores. Every run is basically a clean slate. 

I wasn't able to reproduce the editing issue this evening, but I did find the bug in the individual bank download and hopefully fixed that. Version 1.0.4 is up, let me know how that treats you.

**Comment by malacalypse** [comment]
Send to active slot on double click is always greyed out, I haven't implemented turning that off, and I left it there in the menu as a reminder that it's possible to do.

**Comment by lorenzolamasse** [comment]
Issue with bank fecthing seems sorted, yeah !
I think I figured out why I can't rename some of the patches, there seem to have special (invisible) characters at the end of these name, as I have to press backspace several times before actually removing the last character, some trailing escape characters when fetching ? As after removing these characters I can edit the name properly.
